---
title: "Barnacles SNP and DE"
author: "Eddy Dowle"
date: "3/6/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Exploring SNP analyses with DE 

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. Y

Bring in DE file. This is just the DE genes

```{r DE,message = FALSE}
library(dplyr)
```

```{r DE_2}
expressiondat<-read.csv("~/Documents/github/barnacles/annot.results.csv",header=T)
colnames(expressiondat)
```

## FST 
Bring in file interested. Here Fst:
(change the # to switch to a stricter set)

```{r DE_FST1}
#fst<-read.table("~/Documents/Barnacles/notreat/popoolation/BCT_WST.max50000min10.fst",header=T)
fst<-read.table("~/Documents/Barnacles/notreat/popoolation/BCT_WST.max50000min50.fst",header=T)
```

Depends what you want to look at. If you have particular scaffolds that 
Just looking at the value of the top snp in a scaffold here just a rough look

```{r DE_FST2}
fst_top <- fst %>% group_by(.,SeqName) %>% slice(which.max(fst_values_BCT.WST))

fst_top %>% filter(., SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,fst_values_BCT.WST) %>% colMeans()
fst_top %>% filter(., !SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,fst_values_BCT.WST) %>% colMeans()
```

Not a lot going on the FST other than that they are really high

##Tajima D

some of these got annotated as having two genes on the same scaffold Im just going to average by scafold ~not sure if that is the 'right' way to go about this

```{r DE_D}

dcorrected_BCT<-read.table("~/Documents/Barnacles/notreat/popoolation/barnacles.BCT.notreat.sort.gtf.min4max50000.downsample.dcorrected.clean",header=T)
dcorrected_WST<-read.table("~/Documents/Barnacles/notreat/popoolation/barnacles.WST.notreat.sort.gtf.min4max50000.downsample.dcorrected.clean",header=T)

dcorrected_BCT<-dcorrected_BCT %>% group_by(.,SeqName) %>% summarise(.,dcorrected_ave = mean(dcorrected, na.rm = T),snp_count=sum(number_snps))
dcorrected_WST<-dcorrected_WST %>% group_by(.,SeqName) %>% summarise(.,dcorrected_ave = mean(dcorrected, na.rm = T),snp_count=sum(number_snps))

dcorrected_BCT %>% filter(., SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
dcorrected_BCT %>% filter(., !SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
dcorrected_WST %>% filter(., SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
dcorrected_WST %>% filter(., !SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
```

More interesting! Definitly something different there!

## FET

Seems to have the same problem as the FST they are just too different
have to drop values that came in as Inf-less than ideal
I think the tajimas D values are going to be most interesting as the differentiation is so high I think it just sweeps any signal out


```{r DE_FET}
fet<-read.table("~/Documents/Barnacles/notreat/popoolation/BCT_WST.maxcov50000min10.fet",header=T)
#fet<-read.table("~/Documents/Barnacles/notreat/popoolation/BCT_WST.maxcov50000min50.fet",header=T)

fet_top <- fet %>% group_by(.,SeqName) %>% filter(., !fisher_values_.log10.p.value._BCT.WST==Inf) %>%  slice(which.max(fisher_values_.log10.p.value._BCT.WST))
fet_top %>% filter(., SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,fisher_values_.log10.p.value._BCT.WST) %>% colMeans()
fet_top %>% filter(., !SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,fisher_values_.log10.p.value._BCT.WST) %>% colMeans()
```

There is also a program out recently that might be of interest:
https://academic.oup.com/nar/article/44/19/e148/2468394