---
title: "Barnacles SNP and DE"
author: "Eddy Dowle"
date: "3/7/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Exploring SNP analyses with DE 

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. Y

Bring in DE file. This is just the DE genes

```{r DE,message = FALSE}
library(dplyr)
```

```{r DE_2}
expressiondat<-read.csv("~/Documents/github/barnacles/annot.results.csv",header=T)
colnames(expressiondat)
```

##Tajima D

#Across all Differentially Expressed (DE) transcripts

Some of these are annotated as having two genes on the same scaffold Im just going to average by scafold ~not sure if that is the 'right' way to go about this

```{r DE_D}

dcorrected_BCT<-read.table("~/Documents/Barnacles/notreat/popoolation/barnacles.BCT.notreat.sort.gtf.min4max50000.downsample.dcorrected.clean",header=T)
dcorrected_WST<-read.table("~/Documents/Barnacles/notreat/popoolation/barnacles.WST.notreat.sort.gtf.min4max50000.downsample.dcorrected.clean",header=T)

dcorrected_BCT<-dcorrected_BCT %>% group_by(.,SeqName) %>% summarise(.,dcorrected_ave = mean(dcorrected, na.rm = T),snp_count=sum(number_snps))
dcorrected_WST<-dcorrected_WST %>% group_by(.,SeqName) %>% summarise(.,dcorrected_ave = mean(dcorrected, na.rm = T),snp_count=sum(number_snps))

#average tajima D within BCT that are DE 
dcorrected_BCT %>% filter(., SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
#average tajima D within BCT that are NOT DE 
dcorrected_BCT %>% filter(., !SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
#average tajima D within WST that are DE
dcorrected_WST %>% filter(., SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
#average tajima D within WST that are NOT DE
dcorrected_WST %>% filter(., !SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
```

BCT: t-test

```{r DE_D2}
a<- dcorrected_BCT %>% filter(., SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,dcorrected_ave)
hist(a$dcorrected_ave, main="BCT DE")
b<-dcorrected_BCT %>% filter(., !SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) 
hist(b$dcorrected_ave,main="BCT not DE")
```

t-test Tajima's D DE transcripts vs not DE transcripts: BCT
```{r DE_D3}
t.test(a$dcorrected_ave,b$dcorrected_ave)
```

WST: t-test

```{r DE_D4}
a<- dcorrected_WST %>% filter(., SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,dcorrected_ave)
hist(a$dcorrected_ave,main="WST DE")
b<-dcorrected_WST %>% filter(., !SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) 
hist(b$dcorrected_ave,main="WST not DE")
```

t-test Tajima's D DE transcripts vs not DE transcripts: WST
```{r DE_D5}
t.test(a$dcorrected_ave,b$dcorrected_ave)
```

#Question One: transcripts that were only differentially regulated in the White Sea (that's the more freeze tolerant) population

```{r DE_D6}
expressionWST<-read.csv("~/Documents/github/barnacles/allwsdiffreg.csv",header=T)

#average tajima D within BCT that are DE in only in white sea
dcorrected_BCT %>% filter(., SeqName %in% expressionWST$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
#average tajima D within BCT that are NOT DE in only in white sea
dcorrected_BCT %>% filter(., !SeqName %in% expressionWST$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
#average tajima D within WST that are DE in only in white sea
dcorrected_WST %>% filter(., SeqName %in% expressionWST$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
#average tajima D within WST that are NOT DE in only in white sea
dcorrected_WST %>% filter(., !SeqName %in% expressionWST$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
```

BCT: t-test

```{r DE_D7}
a<- dcorrected_BCT %>% filter(., SeqName %in% expressionWST$SeqName) %>% ungroup() %>% select(.,dcorrected_ave)
hist(a$dcorrected_ave,main="BCT DE (in WST)")

b<-dcorrected_BCT %>% filter(., !SeqName %in% expressionWST$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) 
hist(b$dcorrected_ave,main="BCT not DE (in WST)")
```

t-test Tajima's D DE transcripts in WST vs not DE transcripts in WST: BCT
```{r DE_D8}
t.test(a$dcorrected_ave,b$dcorrected_ave)
```

WST: t-test: 

```{r DE_D9}
a<- dcorrected_WST %>% filter(., SeqName %in% expressionWST$SeqName) %>% ungroup() %>% select(.,dcorrected_ave)
hist(a$dcorrected_ave,main="WST DE (in WST)")

b<-dcorrected_WST %>% filter(., !SeqName %in% expressionWST$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) 
hist(b$dcorrected_ave,main="WST not DE (in WST)")
```
t-test Tajima's D DE transcripts in WST vs not DE transcripts in WST: WST
```{r DE_D10}
t.test(a$dcorrected_ave,b$dcorrected_ave)
```

#Question Two: Anything labelled with "transcription" as a function.  I think these are likely transcription factors, and in particular, XLOC_006827 and XLOC_01766 are upregulated in both the BC and WS populations in the 4 hours following freezing

```{r DE_D11}
expressionWST_transciption<-expressionWST %>% filter(.,grepl('transcription',InterPro.GO.Names) | grepl('transcription',GO.Names)) 

#average tajima D within BCT that are DE in only in white sea with a GO term of 'transciption'
dcorrected_BCT %>% filter(., SeqName %in% expressionWST_transciption$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
#average tajima D within BCT that do not have a GO term of 'transciption'
dcorrected_BCT %>% filter(., !SeqName %in% expressionWST_transciption$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
#average tajima D within WST that are DE in only in white sea with a GO term of 'transciption'
dcorrected_WST %>% filter(., SeqName %in% expressionWST_transciption$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
#average tajima D within WST that do not have a GO term of 'transciption'
dcorrected_WST %>% filter(., !SeqName %in% expressionWST_transciption$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
```

There isnt enough of these transcipts in the BCT population for a t-test
```{r DE_D12}
a<- dcorrected_BCT %>% filter(., SeqName %in% expressionWST_transciption$SeqName) %>% ungroup() %>% select(.,dcorrected_ave)
hist(a$dcorrected_ave,main="BCT DE transcription transcripts (in WST)")
nrow(a)
b<-dcorrected_BCT %>% filter(., !SeqName %in% expressionWST_transciption$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) 
hist(b$dcorrected_ave,main="BCT not DE transcription transcripts (in WST)")
nrow(b)
#t.test(a$dcorrected_ave,b$dcorrected_ave)
```

WST: t-test: 

```{r DE_D13}
a<- dcorrected_WST %>% filter(., SeqName %in% expressionWST_transciption$SeqName) %>% ungroup() %>% select(.,dcorrected_ave)
hist(a$dcorrected_ave)
nrow(a)
b<-dcorrected_WST %>% filter(., !SeqName %in% expressionWST_transciption$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) 
hist(b$dcorrected_ave)
nrow(b)
```
t-test Tajima's D DE transcripts with transciption in GO term in WST vs not DE transcripts in WST: WST
```{r DE_D14}
t.test(a$dcorrected_ave,b$dcorrected_ave)
```

#Question Three: Anything labelled as a potential aquaporin. (XLOC_019964, XLOC_019967, and XXLOC_035387 in particular)

Only three transcipts return from the search terms aquaporin or Aquaporin so not running t-tests
```{r DE_D20}
expressionWST_aqua <-expressionWST %>% filter(.,grepl('aquaporin',Description)|grepl('Aquaporin',Description)) 

dcorrected_BCT %>% filter(., SeqName %in% expressionWST_aqua$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
#no transcripts
#average tajima D within BCT that do not have a description of 'Aquaporin'
dcorrected_BCT %>% filter(., !SeqName %in% expressionWST_aqua$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
#average tajima D within WST that are DE in only in white sea with description of 'Aquaporin'
dcorrected_WST %>% filter(., SeqName %in% expressionWST_aqua$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
#average tajima D within WST that do not have a description of 'Aquaporin'
dcorrected_WST %>% filter(., !SeqName %in% expressionWST_aqua$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
```


#Question Four: Anything labelled as a "macrophage mannose receptor".  I think these are potentially novel antifreeze proteins--they often blast to fish Type 2 antifreeze proteins

Note: there is only one transcipt in this comparison though for BCT coverage isnt high enough for these transcripts for a Tajima D calculation
```{r DE_D15}
expressionWST_MMR <- expressionWST %>% filter(.,grepl('macrophage mannose receptor',Description)) 

#average tajima D within BCT that are DE in only in white sea with description of 'macrophage mannose receptor'
dcorrected_BCT %>% filter(., SeqName %in% expressionWST_MMR$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
#average tajima D within BCT that do not have a description of 'macrophage mannose receptor'
dcorrected_BCT %>% filter(., !SeqName %in% expressionWST_MMR$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
#average tajima D within WST that are DE in only in white sea with description of 'macrophage mannose receptor'
dcorrected_WST %>% filter(., SeqName %in% expressionWST_MMR$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
#average tajima D within WST that do not have a description of 'macrophage mannose receptor'
dcorrected_WST %>% filter(., !SeqName %in% expressionWST_MMR$SeqName) %>% ungroup() %>% select(.,dcorrected_ave) %>% colMeans()
```
there isnt enough of the MMR transcripts to do t-tests

## FST 
Bring in file interested. Here Fst:
(change the # to switch to a stricter set)

```{r DE_FST1}
#fst<-read.table("~/Documents/Barnacles/notreat/popoolation/BCT_WST.max50000min10.fst",header=T)
fst<-read.table("~/Documents/Barnacles/notreat/popoolation/BCT_WST.max50000min50.fst",header=T)
```

Depends what you want to look at. If you have particular scaffolds that 
Just looking at the value of the top snp in a scaffold here just a rough look

```{r DE_FST2}
fst_top <- fst %>% group_by(.,SeqName) %>% slice(which.max(fst_values_BCT.WST))

fst_top %>% filter(., SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,fst_values_BCT.WST) %>% colMeans()
fst_top %>% filter(., !SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,fst_values_BCT.WST) %>% colMeans()
```

Not a lot going on the FST other than that they are really high


## FET

Seems to have the same problem as the FST they are just too different
have to drop values that came in as Inf-less than ideal
I think the tajimas D values are going to be most interesting as the differentiation is so high I think it just sweeps any signal out


```{r DE_FET}
fet<-read.table("~/Documents/Barnacles/notreat/popoolation/BCT_WST.maxcov50000min10.fet",header=T)
#fet<-read.table("~/Documents/Barnacles/notreat/popoolation/BCT_WST.maxcov50000min50.fet",header=T)

fet_top <- fet %>% group_by(.,SeqName) %>% filter(., !fisher_values_.log10.p.value._BCT.WST==Inf) %>%  slice(which.max(fisher_values_.log10.p.value._BCT.WST))
fet_top %>% filter(., SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,fisher_values_.log10.p.value._BCT.WST) %>% colMeans()
fet_top %>% filter(., !SeqName %in% expressiondat$SeqName) %>% ungroup() %>% select(.,fisher_values_.log10.p.value._BCT.WST) %>% colMeans()
```

There is also a program out recently that might be of interest:
https://academic.oup.com/nar/article/44/19/e148/2468394